lib/matplotlib/tests/test_datetime.py

```diff
@@ -77,11 +78,38 @@ def test_axhspan(self):
         ax3.set_xlabel('Date')
         ax3.set_ylabel('Date')
 
-    @pytest.mark.xfail(reason="Test for axline not written yet")
     @mpl.style.context("default")
     def test_axline(self):
-        fig, ax = plt.subplots()
-        ax.axline(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        fig, (ax1, ax2, ax3) = plt.subplots(3, 1,
+                                            constrained_layout=True,
+                                            figsize=(10, 12))
+
+        start_date = datetime.datetime(2023, 1, 1)
+        dates = [start_date + datetime.timedelta(days=i) for i in range(31)]
+        values = list(range(1, 32))
+
+        ax1.plot(dates, values, 'ro')
+        ax1.axline((date2num(dates[0]), values[0]), (date2num(dates[-1]), values[-1]),
```

Comment:
I did the minimal change here to get the point-point version working with the default transform (data space). I do think that the final version would need at least some more.
 - passing `kwargs` in
 - Maybe pushing the unit transforming down into `mlines.AxLine`? (trying to trace how base `Line2D` does it, if that does it in the class, I'd probably follow suit for `AxLine`, but not super clear on that, actually... `AxLine` inherits from `Line2D`)
 - some more thought about `slope` may be good... it is already disallowed with non-linear scales, may be best to just disallow it with unitful data or declare it to be in explicitly "unitless data space" (currently it states that it is in 'data coordinates'). I think it may actually work out for at least the case of having the same units on both axes (i.e. the slope is unitless then) or possibly also with full unit libraries, though possibly would require pushing units into the `AxLine` class for it to work.